### ISSUE TITLE
Add Powerdns Recursor input plugin

### ISSUE THREAD

**Issue by descrepes** [influxdata/telegraf#812]
Hi,

The actual Powerdns input plugin seems to be available for the "pdns_server" socket only.
It would be awesome if it could collect from the "pdns_recursor" socket :p

Thx a lot !

**Comment by alexzorin** [comment]
+1 would be nice.

`rec_control` in PowerDNS Recursor 4.x appears to use unix datagram socket that I can't quite wrap my head around from Go (unixgram times out, need to look at rec_control implementation more closely):

https://github.com/PowerDNS/pdns/blob/09cc8c7a36414ed93dd3b67c4767298da7c2f10d/pdns/rec_channel.cc#L60

```
socket(PF_LOCAL, SOCK_DGRAM, 0)         = 3
fcntl(3, F_GETFD)                       = 0
fcntl(3, F_SETFD, FD_CLOEXEC)           = 0
setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
gettimeofday({1473131680, 118010}, NULL) = 0
open("/var/run/lsocknyxXzx", O_RDWR|O_CREAT|O_EXCL, 0600) = 4
unlink("/var/run/lsocknyxXzx")          = 0
bind(3, {sa_family=AF_LOCAL, sun_path="/var/run/lsocknyxXzx"}, 110) = 0
chmod("/var/run/lsocknyxXzx", 0666)     = 0
connect(3, {sa_family=AF_LOCAL, sun_path="/var/run/pdns_recursor.controlsocket"}, 110) = 0
sendto(3, "get-all", 7, 0, NULL, 0)     = 7
...
```

and also the output is with new lines and field delimited by whitespace:

```
unauthorized-tcp        0
unauthorized-udp        0
unexpected-packets      0
unreachables    0
uptime  4625
user-msec       209
```

**Comment by ahupowerdns** [comment]
Hi - there is an API to retrieve these metrics now, please see https://doc.powerdns.com/authoritative/common/api/endpoint-statistics.html
Is that helpful?
This works on all PowerDNS products.

**Comment by mikalsande** [comment]
I needed to get metrics from pdns recursor into influx and found this post. Using the API works well with the exec input. Its not pretty, but it works. I even got my Lua counters in there :)


```
[[inputs.exec]]
  commands = [ "/bin/sh /etc/telegraf/pdns-recursor.sh" ]
  timeout = "5s"
  name_suffix = "_pdns"
  data_format = "json"
```


/etc/telegraf/pdns-recursor.sh
`/bin/curl -s -H 'X-API-Key: <API key>' http://127.0.0.1:8082/api/v1/servers/localhost/statistics | jq 'reduce .[] as $item ({}; . + { ($item.name): ($item.value|tonumber)})'`

**Comment by dupondje** [comment]
Merge request open for this:
https://github.com/influxdata/telegraf/pull/4545

**Comment by danielnelson** [comment]
Closed in #4545

**Comment by dmgeurts** [comment]
> /etc/telegraf/pdns-recursor.sh `/bin/curl -s -H 'X-API-Key: <API key>' http://127.0.0.1:8082/api/v1/servers/localhost/statistics | jq 'reduce .[] as $item ({}; . + { ($item.name): ($item.value|tonumber)})'`

The above no longer works due to nested data, the following works for me (and can probably be improved upon):

curl -s -H 'X-API-Key: <API key>' http://127.0.0.1:8082/api/v1/servers/localhost/statistics | jq 'reduce .[] as $item ({}; . + { ($item.name): ($item.value)})' | jq 'to_entries | map(if (.value | type) == "string" then .value |= tonumber else .key as $key | .value[] | .name |= $key+"."+. | .value |= tonumber end) | from_entries'
